lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -62,75 +66,179 @@ def __init__(self, config):
+                  + "a workspace ID over a mutable workspace name that may cause race conditions")
+            ws_id = self.dfu.ws_name_to_id(ws_name)
+        mass_params = {_WSID: ws_id, _INPUTS: [inputs]}
+        return mass_params
+
+    def _validate_mass_params(self, params):
+        ws_id = self._get_int(params.get(_WSID), _WSID)
+        if not ws_id:
+            raise ValueError(f"{_WSID} is required")
+        inputs = params.get(_INPUTS)
+        if not inputs or type(inputs) != list:
+            raise ValueError(f"{_INPUTS} field is required and must be a non-empty list")
+        for i, inp in enumerate(inputs, start=1):
+            if type(inp) != dict:
+                raise ValueError(f"Entry #{i} in {_INPUTS} field is not a mapping as required")
+            self._validate_params(inp)
+
+    def _get_int(self, putative_int, name, minimum=1):
+        if putative_int is not None:
+            if type(putative_int) != int:
+                raise ValueError(f"{name} must be an integer, got: {putative_int}")
+            if putative_int < minimum:
+                raise ValueError(f"{name} must be an integer >= {minimum}")
+        return putative_int
+
+    def _import_genbank_mass(self, params):
+
+        workspace_id = params[_WSID]
+        inputs = params[_INPUTS]
+
+        genome_names = []
+        genome_data = []
+        genome_meta = []
+
+        genome_objs = []
+        consolidated_files = []
+        input_directories = []
+
+        for idx, input_params in enumerate(inputs):
+
+            genome_obj = _Genome(self.cfg)
+            genome_names.append(input_params['genome_name'])
+
+            # construct the input directory staging area
+            input_directory = self._stage_input(input_params)
+            input_directories.append(input_directory)
+
+            # update default params
+            input_params = {**genome_obj.default_params, **input_params}
+            genome_meta.append(input_params['metadata'])
+            inputs[idx] = input_params
+
+            genome_obj.generate_parents = input_params.get('generate_missing_genes')
+            genome_obj.generate_ids = input_params.get('generate_ids_if_needed')
+            if input_params.get('genetic_code'):
+                genome_obj.code_table = input_params['genetic_code']
+            genome_objs.append(genome_obj)
+
+            # do the upload
+            files = self._find_input_files(input_directory)
+            consolidated_file = self._join_files_skip_empty_lines(files)
+            consolidated_files.append(consolidated_file)
+
+        # save files to shock mass
+        file_handles = self._save_files_to_blobstore(consolidated_files)
+
+        # write and save assembly file
+        assemblies_ref = self._save_assemblies(
+            workspace_id, consolidated_files, inputs, genome_objs
+        )
+
+        # get assembly data
+        assemblies_data = self._get_assemblies_data(assemblies_ref)
```

Comment:
This is pulling the data from the workspace that was just saved to the workspace, which seems like a waste of time / bandwidth

Reply:
we need to get `gc_content`, `dna_size`, and `md5` to parse in _parse_genbank func. If we don't get this info from `dfu`, where could we get them?

Reply:
Hmm, I guess assembly util doesn't return those. Two options then, in order of goodness but reverse order of work:

1). Update assembly util to return the info we need (and maybe any other small data that gets calculated)
2) At least make sure the calls to the workspace a) are batched, b) only pull the object_info and not all the data, which is much more expensive

Reply:
@Tianhao-Gu fixed it: https://github.com/kbaseapps/AssemblyUtil/pull/98

Reply:
:+1: